sniffles/feature.py

```diff
@@ -73,13 +108,15 @@ def __str__(self):
 
     def toString(self):
         num_elements = random.randint(2, self.max_list_size)
-        if num_elements > (self.upper_bound - self.lower_bound):
-            print("Error!!!!!")
+        num_elements = min(num_elements,
+                           self.upper_bound - self.lower_bound + 1
+                           )
         myelements = []
-        while len(myelements) < num_elements:
-            mypick = random.randint(self.lower_bound, self.upper_bound)
-            if mypick not in myelements:
-                myelements.append(mypick)
+        for i in range(self.lower_bound, self.upper_bound + 1):
```

Comment:
This solution is simpler in that it removes the need to keep track of elements already picked.  However, it is less efficient in that it requires building the list of all possible elements prior to picking some elements.  This will work with one and two byte integers, but probably will not work with 4 byte or larger integers, which are possible boundaries.